### PR TITLE
feat(cli): upgrade aws-sdk to sdkv3

### LINKED ISSUE #25870: (cli): (cdk fails with "Profile <profile name> did not include credential process)
### Describe the bug

We want to use the CDK CLI with Gitlab Runner, which is running on EKS in Account A. **cdk deploy** is executed for Account B (Cross-Account Deployment).

```
cdk deploy --profile AccountB
```

CLI Command fails with:
```
Unable to determine the default AWS account (ProcessCredentialsProviderFailure): Profile <AccountB> did not include credential process
....
Could not assume arn:aws:iam::xxxxxxx:role/cdk-xxxxxxxx-lookup-role-xxxxxxx-eu-central-1, proceeding anyway.
```

We are using IRSA (IAM Role for Service Accounts), further called **Role A,** for the Gitlab Runner in **Account A**. 
In Account B there is an IAM Role (**Role B**), which can be assumed by **Role A**.  **RoleB** can assume all cdk-provisioned IAM Roles in that Account (created by cdk bootstrap).

Profile file (stored in /configmaps/awsconfig):
```
    [default]
    region = eu-central-1
    role_arn = arn:aws:iam::AccountA:role/RoleA
    web_identity_token_file=/var/run/secrets/eks.amazonaws.com/serviceaccount/token
    output = json


    [profile AccountB]
    role_arn = arn:aws:iam::AccountB:role/RoleB
    source_profile = default
    output = json
    region = eu-central-1
```

We have set environment variables **AWS_PROFILE** and **AWS_CONFIG_FILE**:
```
AWS_PROFILE=AccountB
AWS_CONFIG_FILE=/configmaps/awsconfig
```

**AWS CLI** is working fine with above profile file and the environment variables:
```
aws sts get-caller-identity
{
    "UserId": "AROA2SFT2XXXXXXXXX:botocore-session-168600000",
    "Account": "00000000000",
    "Arn": "arn:aws:sts::00000000000:assumed-role/RoleB/botocore-session-168600000"
}
```

Assuming the cdk-lookup Role from **RoleB** with **AWS CLI** works fine.
```
aws sts assume-role --role-arn arn:aws:iam::xxxxxxx:role/cdk-xxxxxxxx-lookup-role-xxxxxxx-eu-central-1 --role-session-name test123
{
    "Credentials": {
        "AccessKeyId": "<somekey>",
        "SecretAccessKey": "<somesecret>",
        "SessionToken": "<nice jwt>",
        "Expiration": "2023-06-06T15:29:36+00:00"
    },
    "AssumedRoleUser": {
        "AssumedRoleId": "<some otherkey:test123",
        "Arn": "arn:aws:iam::xxxxxxx:role/cdk-xxxxxxxx-lookup-role-xxxxxxx-eu-central-1"
    }
}
```

By the way: cdk also seems not to respect **AWS_PROFILE** environment variable as it is using the default role (it could be, that the session token by IRSA is used), when NOT using **--profile**.

### Expected Behavior

We expect that **AWS CDK CLI** is acting identically to **AWS CLI** when using AWS profiles for Cross-Account Deployments. Environment Variable **AWS_PROFILE** should be respected.

### Current Behavior

The CDK CLI fails that the cdk-Lookup-Role cannot be assumed, because "Profile <AccountB> did not include credential process". When you don't specify **--profile**, then default-Profile (or session token created by OIDC Provider of IAM) is used.

### Reproduction Steps

You need to run a pod in EKS:

- configure IRSA (RoleA) for the pod in the same account (Account A)
- configure a RoleB in Account B, which can be assumed by RoleA
- give permissions to RoleB to assume cdk-provisioned roles
- configure an AWS_CONFIG_FILE like described above.
- install CDK, NODE etc on the pod
- Configure env variables **AWS_PROFILE** and **AWS_CONFIG_FILE**
- Run cdk deploy

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.78.0

### Framework Version

2.78.0

### Node.js Version

v18.16.0

### OS

node:18-alpine

### Language

Typescript

### Language Version

3.9.7

### Other information

_No response_

### PR DESCRIPTION
**Pull request by TheRealAmazonKendra**
This PR updates the CDK CLI to use the AWS SDK V3 instead of V2. 

### Manual Test Cases for Authorization
All tests were run verbosely so that I could manually check the credentials being used from the CLI output.

- [x] No credentials setup and no default profile fails as expected (established to ensure nothing was unintentionally setup)
- [x] Explicitly provided profile sourced from config file, tested with both `--profile` and `AWS_PROFILE`
  - [x] `aws_access_key_id` and `aws_secret_access_key`
  - [x] `aws_access_key_id` and `aws_secret_access_key` and `region`
  - [x] `source_profile` and `role_arn` that points to another profile with `aws_access_key_id` and `aws_secret_access_key` in both config and credentials
  - [x] `source_profile` and `role_arn` that points to Environment
  - [x] `source_profile` that and `role_arn` that points to SSO profile in both config and credentials
  - [x] SSO both using all fields in `[profile]` and split between `[sso-session]` and `[profile]`
  - [x] `credential_source` is Environment
  - [x] `credential_process`
- [x] Explicitly provided profile sourced from  credentials file
  - [x] `aws_access_key_id` and `aws_secret_access_key`
  - [x] `aws_access_key_id` and `aws_secret_access_key` and `region`
  - [x] `source_profile` and `role_arn` that points to another profile with `aws_access_key_id` and `aws_secret_access_key` in both config and credentials
  - [x] `source_profile` and `role_arn` that points to Environment
  - [x] `source_profile` that and `role_arn` that points to SSO profile in both config and credentials
  - [x]  `credential_source` is Environment
  - [x] `credential_source` is EcsContainer (integ tests in CodeBuild)
  - [x] `credential_process`
- [x] Default profile (in both config file and credentials file)
  - [x] `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment variables set (cleared after test)
  - [x] `AMAZON_ACCESS_KEY_ID` and `AMAZON_SECRET_ACCESS_KEY` environment variables set (cleared after test) 
  - [x] `aws_access_key_id` and `aws_secret_access_key`
  - [x] `aws_access_key_id` and `aws_secret_access_key` and `region`
  - [x] `source_profile` and `role_arn` that points to a profile with `aws_access_key_id` and `aws_secret_access_key`
  - [x] `source_profile` and `role_arn` that points to Environment (this will use `fromEnv` prior to getting to looking for `source_profile` in the Ini file)
  - [x] SSO profile is setup as default 
  - [x] `credential_process`

Missing cases:
- Web Identity Token
- Ec2Instance as `credential_source`
- Metadata Service
- MFA

Closes #25870, #26292, #20956, #24744, #27265, 20896.

### Reason for this change

The AWS SDK V2 is now in maintenance mode.

***Note: This is now blocked by https://github.com/aws/aws-cdk/pull/31623 because the underlying changes to cdk-assets were not merged to the v3 branch so that will need to be done before this can be rebased and merged.

Additionally, the version will need to be updated for skipping regression tests when this PR is ready to be merged.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk/jest.config.js [hunk 1]
```diff
@@ -7,12 +7,18 @@ module.exports = {
             // DO NOT LOWER THESE VALUES!
             // If you need to break glass, open an issue to re-up the values with additional test coverage
             statements: 81,
-            branches: 68,
+            branches: 66,
```
Comment (mrgrain) [inline packages/aws-cdk/jest.config.js:10]: Pretty rude 😂

File: packages/aws-cdk/jest.config.js [hunk 2]
```diff
@@ -7,12 +7,18 @@ module.exports = {
...
     },
+    "coveragePathIgnorePatterns": [
+        // Mostly wrappers around the SDK, which get mocked in unit tests
+        "<rootDir>/lib/api/aws-auth/sdk.ts",
+    ],
 
     // We have many tests here that commonly time out
     testTimeout: 30_000,
+    // These tests are too chatty. Shush.
+    silent: true,
```
Comment (mrgrain) [inline packages/aws-cdk/jest.config.js:23]: <3

File: packages/aws-cdk/jest.config.js (hunk 1 above)
Comment (otaviomacedo) [reply packages/aws-cdk/jest.config.js:10]: Lol, and this is was even after writing more tests.

File: packages/@aws-cdk-testing/cli-integ/lib/integ-test.ts [hunk 3]
```diff
@@ -2,10 +2,11 @@ import * as fs from 'fs';
 import * as path from 'path';
 import { MemoryStream } from './corking';
 
-const SKIP_TESTS = fs.readFileSync(path.join(__dirname, '..', 'skip-tests.txt'), { encoding: 'utf-8' })
+const SKIP_TESTS = fs
+  .readFileSync(path.join(__dirname, '..', 'skip-tests.txt'), { encoding: 'utf-8' })
```
Comment (rix0rrr) [inline packages/@aws-cdk-testing/cli-integ/lib/integ-test.ts:6]: ☹️  on all these formatting changes

**Review by rix0rrr — APPROVED** [review]
I trust the construction of this nuclear power plant.

**Pull request merged**